src/lib.rs

```diff
@@ -751,6 +846,34 @@ pub unsafe trait FromZeroes {
     /// Self::new_zeroed()`, it differs in that `zero` does not semantically
     /// drop the current value and replace it with a new one - it simply
     /// modifies the bytes of the existing value.
+    ///
+    /// ```
+    /// # mod derive { pub use zerocopy_derive::FromZeroes; }
+    /// # use zerocopy::FromZeroes;
+    /// # use derive::*;
+    /// #
+    /// # #[derive(FromZeroes)]
+    /// # #[repr(C)]
+    /// # struct PacketHeader {
+    /// #     src_port: [u8; 2],
+    /// #     dst_port: [u8; 2],
+    /// #     length: [u8; 2],
+    /// #     checksum: [u8; 2],
+    /// # }
```

Comment:
Do we maybe want to include this in the example so that users see the definition of the type?

Reply:
This example hides the definition itself for brevity, as it's apparent from its initialization:
```rust
let mut header = PacketHeader {
     src_port: 100u16.to_be_bytes(),
     dst_port: 200u16.to_be_bytes(),
     length: 300u16.to_be_bytes(),
     checksum: 400u16.to_be_bytes(),
};
```

In the other example, I *don't* hide the definition, because that example doesn't use the implicit constructor.